### ISSUE TITLE
The process '/opt/hostedtoolcache/goreleaser-action/1.15.2/x64/goreleaser' failed with exit code 1

### ISSUE THREAD

**Issue by charliecon** [goreleaser/goreleaser-action#394]
Hi all!

The release command has been failing consistently for me. The only changes I have made to the yaml file were updating `--rm-dist` to `--clean` and `ubuntu-latest` to `ubuntu-20.04`. The latter was done to fix this bug  - [https://github.com/actions/runner-images/issues/6709#issuecomment-1361269639](https://github.com/actions/runner-images/issues/6709#issuecomment-1361269639). 

The build seems to be reaching the context deadline of 30 minutes. Extending the deadline to one hour did not solve the problem as that deadline was then met. The average time took to release previously was ~15 minutes.

The release failing: [https://github.com/MyPureCloud/terraform-provider-genesyscloud/actions/runs/4163728547/jobs/7205763125](https://github.com/MyPureCloud/terraform-provider-genesyscloud/actions/runs/4163728547/jobs/7205763125)

The last successful release: [https://github.com/MyPureCloud/terraform-provider-genesyscloud/actions/runs/3676626864/jobs/6217514734](https://github.com/MyPureCloud/terraform-provider-genesyscloud/actions/runs/3676626864/jobs/6217514734)

Thanks in advance for any help.
Charlie

**Comment by charliecon** [comment]
For more context, I also re-ran the action with `goreleaser check --debug` which looked like this: [https://github.com/MyPureCloud/terraform-provider-genesyscloud/actions/runs/4165128743/jobs/7207698207](https://github.com/MyPureCloud/terraform-provider-genesyscloud/actions/runs/4165128743/jobs/7207698207)

**Comment by caarlos0** [comment]
goreleaser is just calling go build, your binaries are probably taking longer to build for some reason (more deps, slower gh action runners, etc...)

have you tried comparing build times of your previous and releases locally?

**Comment by caarlos0** [comment]
this last error I don't know, maybe OOMKilled? 

I'm building it locally, it's using all my 12 cpu cores, all my 64gb of ram, and all my 20gb of swap

<img width="1509" alt="CleanShot 2023-02-14 at 12 15 04@2x" src="https://user-images.githubusercontent.com/245435/218779159-3d733df5-f363-456f-88a8-6d8938413d7c.png">

and its definitely taking a long time (>10 min so far), it might be fast for you because you have it cached or something... but it looks like something in your program, or a bug in the go compiler, what could it possibly be I don't know...

**Comment by caarlos0** [comment]
my PC literally died trying to compile this.

tl;dr: its not goreleaser, it merely calls go build. Maybe reducing parallelism helps (`goreleaser release -h`)... 

I don't know what your code and/or deps are doing, but something is funky there.

**Comment by caarlos0** [comment]
would also recommend to clean up your GOMODCACHE and GOCACHE

maybe GOPATH too

**Comment by charliecon** [comment]
It seems the latest version of an sdk we use in the project got too big for github actions to handle. Reducing parallelism seems to have fixed the problem. 

Apologies and thanks for the help Carlos!
